numba/tests/test_literal_dispatch.py

```diff
@@ -326,5 +330,72 @@ def foo(a, b):
                       str(raises.exception))
 
 
+class TestLiteralDispatchWithCustomType(TestCase):
+    def make_dummy_type(self):
+        class Dummy:
+            def lit(self, a):
+                return a
+
+        class DummyType(numba.types.Type):
+            def __init__(self):
+                super().__init__(name="dummy")
+
+        @register_model(DummyType)
+        class DummyTypeModel(models.StructModel):
+            def __init__(self, dmm, fe_type):
+                members = []
+                super().__init__(dmm, fe_type, members)
+
+        @intrinsic
+        def init_dummy(typingctx):
+            def codegen(context, builder, signature, args):
+                dummy = cgutils.create_struct_proxy(
+                    signature.return_type)(context, builder)
+
+                return dummy._getvalue()
+
+            sig = signature(DummyType())
+            return sig, codegen
+
+        @overload(Dummy)
+        def dummy_overload():
+            def ctor():
+                return init_dummy()
+
+            return ctor
+
+        return (DummyType, Dummy)
```

Comment:
not for now, but someday this would be really useful as a utility

Reply:
I was thinking about the same thing. There ought to be a way to make these single-serving custom types.